list/directory/list.js

```diff
@@ -0,0 +1,479 @@
+            let availableLetters = this._getAvailableLetters(dataToUse);
+            let groupToDisplay = this._getGroupFromData(dataToUse, letter);
+            this.setState({data: groupToDisplay, currentSelectedLetter: letter, availableLetters: availableLetters});
+        }
+
+    },
+
+    /**
+     * Filter results by searching text in data.
+     * @private
+     */
+    _filterResults(){
+        this.searchText = this.quickSearchDOMNode.value.trim();
+        let newState = this._getState();
+
+        this.setState(
+            {   data: newState.data,
+                currentSelectedLetter: newState.currentSelectedLetter,
+                availableLetters: newState.availableLetters
+            });
+    },
+
+    /**
+     * Get group size.
+     * @param {string} selected letter
+     * @returns {int} group size
+     * @private
+     */
+    _getGroupSize(letter){
+        if(this.state && this.state.data && isArray(this.state.data[letter])){
+            return this.state.data[letter].length;
+        }
+    },
+
+    /**
+     * Render letters filter.
+     * @returns {JSX} Htm content.
+     * @private
+     */
+    _renderLettersFilter(){
+        return (
+            <LettersFilter
+                selectedLetter={this.state.currentSelectedLetter}
+                availableLetters={this.state.availableLetters}
+                handleLetterSelection={this._onLetterSelected}
+                extraCharsSymbol={nonAlphabeticCharactersSymbol}
+                />
+        );
+    },
+
+    /**
+     * Render quick search.
+     * @returns {JSX} Htm content.
+     * @private
+     */
+    _renderQuickSearch(){
+        if(document.querySelector(this.props.externalQuickSearch) == null){
+            return(
+                <div style={{display: 'inline-block'}}>
+                    <input ref="quickSearch" className="form-control" type="text" placeholder="search text"></input>
+                </div>
+            );
+        }
+    },
+
+    /**
+     * Render group header.
+     * @param {string} selected letter
+     * @returns {JSX} Htm content.
+     * @private
+     */
+    _renderGroupHeader(letter){
+        return(
+            <div data-focus='directory-list-group'>
+                <div data-focus="directory-list-group-letter" className="circle">{letter}</div>
+                {this.props.displayGroupSize ?  <span data-focus="directory-list-group-size"> ({this._getGroupSize(letter)})</span>: false}
+            </div>
+        );
+    },
+
+    /**
+     * Render data list.
+     * @returns {JSX} Htm content.
+     * @private
+     */
+    _renderDataList(){
+        let listProps = omit(this.props, ['fieldName', 'showAll', 'displayGroupSize', 'data', 'externalQuickSearch', 'withQuickSearch', 'globalSearch', 'doAfterSearch', 'displayGroupHeader']);
+
+        if(this.props.data && isEmpty(this.props.data)){
+            return(
+                <div>{MESSAGE_EMPTY_LIST}</div>
```

Comment:
your messages needs to be translated, your component sould have `i18nmixin` and perform a `this.i18n`